### DISCUSSION TITLE
Format Errors Thrown by Zod Schemas

### DISCUSSION THREAD

**Discussion by patrickkabwe** [trpc/trpc#3857] [Q&A]
Is there a way to format zod errors to your liking?

//Router
```
export const authRouters = t.router({
  login: t.procedure.input(loginSchema).mutation(loginResolver),
})
```

//Schema
```
import { z } from "zod";

const loginSchema = z.object({
  email: z.string().email(),
  password: z.string().min(8),
});

```

//Resolver
```
export const loginResolver: ResolverHandler<LoginPayload> = async ({
  input,
  ctx,
}) => {
  const authService = new AuthService();
  const user = await authService.login(ctx.prisma, input);
  return user;
};
```

What I need is to catch the validation errors that zod return if input is not valid and throw a custom error

I have tried to use errorFormatter doesn't seem to work (not even called)

Your help will be appreciated.

**Comment by KATT** [comment]
Error formatting should work https://trpc.io/docs/error-formatting
